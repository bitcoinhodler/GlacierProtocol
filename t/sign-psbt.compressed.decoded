{
  "txid": "d0f7af332d20477133fa88900c86a62cce461dcffa641f53996527fff15b25ea",
  "hash": "d0f7af332d20477133fa88900c86a62cce461dcffa641f53996527fff15b25ea",
  "version": 2,
  "size": 405,
  "vsize": 405,
  "weight": 1620,
  "locktime": 0,
  "vin": [
    {
      "txid": "46d252dc233a84211980ac834246048fb8589fdf6794d82993a56a710a7c77f9",
      "vout": 1,
      "scriptSig": {
        "asm": "0 3044022029a967f9ab890bf67e2014c9260382fd1500745c931f5a4967131958b5a85a2e02202c3fd6bd1ee35743e1a9fea33defb9e365d59596e6254cb39d36e2987cb7d734[ALL] 304402206da7e139c7d2ff18331e9a137d7f8c5ee99c64da4e557186ceba920b627b630202202a408e10d326e0086e0492321aee2c9f37ffde5093ae0bd44f55582cc9aa226b[ALL] 52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae",
        "hex": "00473044022029a967f9ab890bf67e2014c9260382fd1500745c931f5a4967131958b5a85a2e02202c3fd6bd1ee35743e1a9fea33defb9e365d59596e6254cb39d36e2987cb7d7340147304402206da7e139c7d2ff18331e9a137d7f8c5ee99c64da4e557186ceba920b627b630202202a408e10d326e0086e0492321aee2c9f37ffde5093ae0bd44f55582cc9aa226b014c8b52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.00002380,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 606fb72624603795d73b911a729d88f95f9b6ac8 OP_EQUAL",
        "hex": "a914606fb72624603795d73b911a729d88f95f9b6ac887",
        "address": "2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL",
        "type": "scripthash"
      }
    },
    {
      "value": 0.00001000,
      "n": 1,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 5d1e4870ab9df56432aa357021fd0a15a7094ad1 OP_EQUALVERIFY OP_CHECKSIG",
        "hex": "76a9145d1e4870ab9df56432aa357021fd0a15a7094ad188ac",
        "address": "mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF",
        "type": "pubkeyhash"
      }
    }
  ]
}
