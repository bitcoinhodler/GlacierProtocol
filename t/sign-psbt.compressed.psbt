cHNidP8BAHUCAAAAAfN2aK1kG3gjXvG3Lr82noTXKkstk+hWrENikQxMo6XwAQAAAAD/////AkwJAAAAAAAAF6kUYG+3JiRgN5XXO5Eacp2I+V+basiH6AMAAAAAAAAZdqkUXR5IcKud9WQyqjVwIf0KFacJStGIrAAAAAAAAQDfAgAAAAEuO2+wSuow+YpVu8TANAOghDV+xIKeJKOHpCohyOSxFQEAAABqRzBEAiBPPLc/eP9UqAUhs2IaRk3r3x3sC/6yFlVMzqZjmLfzGAIgGxzR+Y3otpSEggqRabBIQe7s8i5lOW8Pmg1O7yZ+SJ4BIQPbXzaJjkGBLS7hXThdASoKCRp9uSvCQoCgxUmj+QIGuf////8CPR/9AAAAAAAZdqkUCJ2yME5IV8NhLSI9IwQ9otNoMXWIrIgTAAAAAAAAF6kUYG+3JiRgN5XXO5Eacp2I+V+basiHAAAAAAEEi1IhAnfxXyKu/68/O8SKKAp2f3xq8hJ2eDwJ/zvKq77OF4ETIQID9kIiO8dR4SWrYE7SDFROkz64KhkgH5C9piNqndvtrCECaJXz7QHAkCW6nPq3tn1pnj66zq2gvv9wvHLH8Q3y0cYhA588sVITDCQeAbkiChR/afcRG4MN7zW1EJc6tHu5r3/2VK4AAQCLUiECd/FfIq7/rz87xIooCnZ/fGryEnZ4PAn/O8qrvs4XgRMhAgP2QiI7x1HhJatgTtIMVE6TPrgqGSAfkL2mI2qd2+2sIQJolfPtAcCQJbqc+re2fWmePrrOraC+/3C8csfxDfLRxiEDnzyxUhMMJB4BuSIKFH9p9xEbgw3vNbUQlzq0e7mvf/ZUrgAA
