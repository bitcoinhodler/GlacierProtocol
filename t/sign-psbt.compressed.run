#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT 2>&1
y
y
y
y
y
y
../../t/sign-psbt.compressed.psbt
y
2
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
cPWVKeCBjnv833Kayc6nczBbG4YqqbCucup3n5D68WPzBEfceBAQ
y
INPUT
