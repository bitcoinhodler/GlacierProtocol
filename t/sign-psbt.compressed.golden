Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
0.00005000 BTC in unspent inputs from cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00002380 BTC going back to cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00001000 BTC going to destination address mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF
Fee amount: 0.00001620
Confirm? (y/n): 
How many private keys will you be signing this transaction with (at least 2 required)?
#: Key #1: Key #2: 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.00005000 BTC in unspent inputs from cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00002380 BTC going back to cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00001000 BTC going to destination address mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF
Fee amount: 0.00001620

Signing with private keys: 
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
cPWVKeCBjnv833Kayc6nczBbG4YqqbCucup3n5D68WPzBEfceBAQ


Confirm? (y/n): 
Calculating transaction...


Raw signed transaction (hex):
0200000001f9777c0a716aa59329d89467df9f58b88f04464283ac801921843a23dc52d24601000000fd1e0100473044022029a967f9ab890bf67e2014c9260382fd1500745c931f5a4967131958b5a85a2e02202c3fd6bd1ee35743e1a9fea33defb9e365d59596e6254cb39d36e2987cb7d7340147304402206da7e139c7d2ff18331e9a137d7f8c5ee99c64da4e557186ceba920b627b630202202a408e10d326e0086e0492321aee2c9f37ffde5093ae0bd44f55582cc9aa226b014c8b52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654aeffffffff024c0900000000000017a914606fb72624603795d73b911a729d88f95f9b6ac887e8030000000000001976a9145d1e4870ab9df56432aa357021fd0a15a7094ad188ac00000000

Transaction fingerprint (md5):
e9c06c24245c26f10234efac08f6f793
QR code for transaction written to transaction.png
